gateway/api_definition.go

```diff
@@ -618,14 +613,40 @@ func (a APIDefinitionLoader) processRPCDefinitions(apiCollection string, gw *Gat
 
 			def.Proxy.ListenPath = newListenPath
 		}
-
-		spec := a.MakeSpec(&def, nil)
-		specs = append(specs, spec)
 	}
+	specs := a.prepareSpecs(apiDefs)
 
 	return specs, nil
 }
 
+func (a APIDefinitionLoader) prepareSpecs(apiDefs []nestedApiDefinition) []*APISpec {
+	var specs []*APISpec
+
+	var wg sync.WaitGroup
```

Comment:
@buger I have something against waitgroups and it's that it can generate a lot of jobs at the same time without a limit. What about if we use a pool of workers? in this case, how many of them we should have? or it's too complex for this problem?

Reply:
https://github.com/golang/go/issues/27837#issuecomment-513443404

Reply:
@titpetric awesome, that leads me to the questions: how many should load at the same time?